santa/src/Santa.Api/Controllers/ClientController.cs

```diff
@@ -930,13 +930,14 @@ public async Task<ActionResult<Logic.Objects.Client>> PutStatus(Guid clientID, [
         public async Task<ActionResult<RelationshipMeta>> UpdateRelationshipStatusByID(Guid clientID, Guid assignmentRelationshipID, [FromBody] EditClientAssignmentStatusModel model)
         {
             BaseClient requestingClient = await repository.GetBasicClientInformationByEmail(User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email).Value);
+            BaseClient baseTargetAgent = await repository.GetBasicClientInformationByID(clientID);
+            List<RelationshipMeta> baseTargetAssignments = await repository.getClientAssignmentInfoByIDAsync(baseTargetAgent.clientID);
 
-            Client targetAgent = await repository.GetClientByIDAsync(clientID);
             AssignmentStatus newAssignmentStatus = await repository.GetAssignmentStatusByID(model.assignmentStatusID);
 
             try
             {
-                RelationshipMeta assignment = targetAgent.assignments.First(a => a.clientRelationXrefID == assignmentRelationshipID);
+                RelationshipMeta assignment = baseTargetAssignments.First(a => a.clientRelationXrefID == assignmentRelationshipID);

```

Comment:
Modified the object to be just their assignments. Should better loading times overall rather than getting a whole client object

Reply:
Originally you got all of their assignments and then iterated through them to check relation IDs.

Now you're getting their basic client information, loading all of their assignments and then iterating through them to check relation IDs.

I'm probably missing something, but how is this a faster operation?

Reply:
Getting a whole client objects gets their assignments AND senders, AND client status among other tables

This way just gets their assignments, because that's all that is needed, hence, faster